src/boss/lambdastore/store.go

```
@@ -0,0 +1,230 @@
+package lambdastore
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/open-lambda/open-lambda/ol/common"
+)
+
+type LambdaStore struct {
+	StorePath string
+	// mapLock protects concurrent access to the Lambdas map
+	mapLock sync.RWMutex
+	Lambdas map[string]*common.LambdaConfig
+	// funcLocks stores a mutex per function to synchronize file-level operations (e.g., upload/delete).
+	funcLocks sync.Map
+}
+
+func NewLambdaStore(storePath string) (*LambdaStore, error) {
+	store := &LambdaStore{
+		StorePath: storePath,
+		Lambdas:   make(map[string]*common.LambdaConfig),
+	}
+
+	if err := os.MkdirAll(store.StorePath, 0755); err != nil {
+		return nil, fmt.Errorf("failed to create lambda store directory: %w", err)
+	}
+
+	files, _ := os.ReadDir(store.StorePath)
```

Let's not ignore errors.